terraform/deployments/cluster-infrastructure/main.tf

```diff
@@ -70,6 +70,13 @@ locals {
       instance_types        = var.arm_workers_instance_types
       update_config         = { max_unavailable = 1 }
       block_device_mappings = local.main_managed_node_group.main.block_device_mappings
+
+      taint = {
+        key = "kubernetes.io/arch"
+        value = "arm64"
+        effect = "NO_SCHEDULE"
```

Comment:
Eeesh, they used different string constants than the ones in the k8s API. Sigh. [Checks out though](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_node_group#taint-configuration-block) 👍